x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Top.java

```
@@ -215,15 +287,20 @@ public AggregatorFunctionSupplier supplier() {
     @Override
     public Expression surrogate() {
         var s = source();
-
-        if (limitValue() == 1) {
-            if (orderValue()) {
-                return new Min(s, field());
-            } else {
-                return new Max(s, field());
+        try {
+            if (limitValue() == 1) {
+                if (orderValue()) {
+                    return new Min(s, field());
+                } else {
+                    return new Max(s, field());
+                }
             }
+        } catch (EsqlIllegalArgumentException e) {
```

Generally I'd expect to only throw if the query is invalid. But it looks like you'd throw if you can't optimize. That feels bad. Can you check if the thing is a `Literal` instead or sometihng?

---

Done